### PR TITLE
Add Github Workflows + clippy fixes

### PR DESCRIPTION
**Pull request by Grinkers**
Workflows before and after clippy fixes. No functionality changes are intended.

before:
https://github.com/Grinkers/edn-rs/actions/runs/5983591578

after:
https://github.com/Grinkers/edn-rs/actions/runs/5983672308

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Check
```
Comment (naomijub) [inline .github/workflows/check.yml]: nice, this has been a pending issue for a while

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+on:
+  push:
+    branches:
+  pull_request:
+    branches:
```
Comment (naomijub) [inline .github/workflows/check.yml]: please use:
```rs
name: Rust

on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "*" ]
```

File: .github/workflows/check.yml (hunk 1 above)
Comment (naomijub) [reply .github/workflows/check.yml]: Standard seems to call it Rust.yml, but this is not an issue

File: .github/workflows/check.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  clippy_check:
+      name: clippy
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v3
+        - uses: dtolnay/rust-toolchain@stable
+          with:
+            components: clippy
+        - run: cargo clippy --all-features -- --deny warnings
```
Comment (naomijub) [inline .github/workflows/check.yml]: We also use pedantic and nursery:
        - run: cargo clippy --all-features -- -W clippy::all -W clippy::pedantic -W clippy::nursery --deny warnings

**Review by naomijub — CHANGES_REQUESTED** [review]
Nice, just a couple of changes

**Comment by Grinkers** [comment]
btw with pedantic and nursey, there's many pages of output. Most of them seem decent, but some are what I consider false positives like
https://rust-lang.github.io/rust-clippy/master/index.html#/must_use_candidate

I'm willing to help maintain and go through everything if #103 can get through. Without clojure.edn/read-string behavior, I can't consider using this crate at this time.

**Comment by naomijub** [comment]
I think I can try to address the remaining issues this week. Thanks

**Pull request merged**